File: 100-Days-Series/Day 63/HourglassPattern.java

public class HourglassPattern {
    // Function to print Hourglass Pattern
    public static void getPrintHourglass(int n){
        // Upper part
        for (int i = n; i >= 1; i--) {
            for (int space = 1; space <= n - i; space++) {
                System.out.print(" ");
            }
            for (int j = 1; j <= (2 * i - 1); j++) {
                System.out.print("*");
            }
            System.out.println();
        }
        // Lower part
        for (int i = 2; i <= n; i++) {
            for (int space = 1; space <= n - i; space++) {
                System.out.print(" ");
            }
            for (int j = 1; j <= (2 * i - 1); j++) {
                System.out.print("*");
            }
            System.out.println();
        }
    }
    // Main Method
    public static void main(String [] args){
        // Let's Start Here
        int rows;
        java.util.Scanner sc=new java.util.Scanner(System.in);
        System.out.print("Enter the number of rows : ");
        rows=sc.nextInt();
        getPrintHourglass(rows);
    }
}

File: 100-Days-Series/Day 63/HourglassPattern.py
# Function to print Hourglass Pattern
def get_print_hourglass(n):
    # Upper Part
    for i in range(n,0,-1):
        for space in range(0,n-i,1):
            print(end=" ")
        for j in range(0,(2*i-1),1):
            print(end="*")
        print()

    # Lower Part
    for i in range(2,n+1,1):
        for space in range(0,n-i,1):
            print(end=" ")
        for j in range(0,(2*i-1),1):
            print(end="*")
        print()

# Let's Start Here
rows=int(input("Enter the number of rows : "))
get_print_hourglass(rows)